diff --git a/package/network/utils/fullconenat-nft/Makefile b/package/network/utils/fullconenat-nft/Makefile
index 7f9276de8c100..5b8a0cc84608e 100644
--- a/package/network/utils/fullconenat-nft/Makefile
+++ b/package/network/utils/fullconenat-nft/Makefile
@@ -15,9 +15,9 @@ PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/fullcone-nat-nftables/nft-fullcone.git
-PKG_SOURCE_DATE:=2023-01-10
-PKG_SOURCE_VERSION:=95ad79bc6d15c64b2770fe8b7092a64d5c2a293c
-PKG_MIRROR_HASH:=37ff8a27ce5e2e38820fba366de30c930d33cf20030e28086778aef62961d335
+PKG_SOURCE_DATE:=2023-05-17
+PKG_SOURCE_VERSION:=07d93b626ce5ea885cd16f9ab07fac3213c355d9
+PKG_MIRROR_HASH:=b2aeb9b804a020504e5a36e0700caf1c860027afb8ffa834573196d96c350598
 
 PKG_LICENSE:=GPL-2.0-only
 PKG_LICENSE_FILES:=LICENSE
