diff --git a/net/curl/Makefile b/net/curl/Makefile
index d2ad55905..7c9f81547 100644
--- a/net/curl/Makefile
+++ b/net/curl/Makefile
@@ -93,7 +93,7 @@ define Package/libcurl
   $(call Package/curl/Default)
   SECTION:=libs
   CATEGORY:=Libraries
-  DEPENDS:= +LIBCURL_WOLFSSL:libwolfssl +LIBCURL_OPENSSL:libopenssl +LIBCURL_GNUTLS:libgnutls +LIBCURL_MBEDTLS:libmbedtls
+  DEPENDS:= +LIBCURL_WOLFSSL:libwolfssl +LIBCURL_OPENSSL:libopenssl +LIBCURL_GNUTLS:libgnutls +LIBCURL_MBEDTLS:libmbedtls +libnghttp3 +libngtcp2
   DEPENDS += +LIBCURL_ZLIB:zlib +LIBCURL_ZSTD:libzstd +LIBCURL_THREADED_RESOLVER:libpthread +LIBCURL_LDAP:libopenldap
   DEPENDS += +LIBCURL_LIBIDN2:libidn2 +LIBCURL_SSH2:libssh2 +LIBCURL_NGHTTP2:libnghttp2 +ca-bundle
   TITLE:=A client-side URL transfer library
@@ -120,6 +120,8 @@ CONFIGURE_ARGS += \
 	--without-ca-path \
 	--without-libpsl \
 	--without-zstd \
+	--with-nghttp3="$(STAGING_DIR)/usr" \
+	--with-ngtcp2="$(STAGING_DIR)/usr" \
 	--with-ca-bundle=/etc/ssl/certs/ca-certificates.crt \
 	\
 	$(call autoconf_bool,CONFIG_IPV6,ipv6) \
diff --git a/net/https-dns-proxy/files/https-dns-proxy.init b/net/https-dns-proxy/files/https-dns-proxy.init
index e403686a6..014e924bc 100755
--- a/net/https-dns-proxy/files/https-dns-proxy.init
+++ b/net/https-dns-proxy/files/https-dns-proxy.init
@@ -150,6 +150,7 @@ start_instance() {
 	append_parm "$cfg" 'logfile' '-l'
 	append_bool "$cfg" 'use_http1' '-x'
 	append_counter "$cfg" 'verbosity' '-v' '0'
+	append_bool "$cfg" 'use_quic_only' '-q'
 
 	procd_open_instance
 # shellcheck disable=SC2086
